alibi/explainers/integrated_gradients.py

```diff
@@ -108,24 +147,184 @@ def _run_forward(model: Union[tf.keras.models.Model],
         Model output or model output after target selection for classification models.
 
     """
+    preds = model(x)
+    if len(model.output_shape) > 1 and model.output_shape[-1] > 1:
+        preds = _select_target(preds, target)
 
-    def _select_target(ps, ts):
-        if ts is not None:
-            if isinstance(ps, tf.Tensor):
-                ps = tf.linalg.diag_part(tf.gather(ps, ts, axis=1))
-            else:
-                raise NotImplementedError
-        else:
-            raise ValueError("target cannot be `None` if `model` output dimensions > 1")
-        return ps
+    return preds
+
+
+def _run_forward_from_layer(model: tf.keras.models.Model,
+                            layer: tf.keras.layers.Layer,
+                            orig_call: Callable,
```

Comment:
btw, is there a reason to define `orig_call` outside this function instead of just getting it inside via `layer.call` ? Is there a risk of `layer.call` already being overwritten somehow?

Reply:
I think it's safer to fix the original layer call method at the very beginning, and stick to that to reset it. The functions may be called many times and in this way we avoid funny things to happen.